packages/sanity/src/desk/panes/document/documentPanel/documentViews/FormView.tsx

```diff
@@ -105,6 +103,12 @@ export function FormView(props: FormViewProps) {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [hasRev])
 
+  const formRef = useRef<null | HTMLDivElement>(null)
+
+  useEffect(() => {
```

Comment:
Yes, it would also work to just have `onSetOpenPath([])` there instead!

However, I think it would be nice to have the focus management logic in one place instead of two. Right now focus management is done both in `FormView` (with `focusFirstDescendant`) and in `DocumentPaneProvider` (with `setFocusPath`). Having this logic in one place makes it easier to maintain going forward, and minimizes the risk of the focus logic in each component interfering with each other.

Using `focusFirstDescendant` will in most cases result in an input being focused, which will result in the presence location being updated. However, if the first focusable element in the form is not an input that is part of the schema, but perhaps some custom element at the top of the form, the presence location will not be updated when the document is opened. In order for the presence location to be updated, an input must be focused in the form that is part of the schema. Therefore, it may be better to use `focusPath` instead of `focusFirstDescendant` to set the initial focus to ensure presence is updated correctly.

Reply:
Okay, I'm convinced 😊 I'll commit the code that you suggested instead then 👍